### PR TITLE
In-memory engine: fix gc may filter keys of onging snapshot

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Close https://github.com/tikv/tikv/issues/17282

What's Changed:

```commit-message
fix gc may filter keys of onging snapshot
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: components/range_cache_memory_engine/src/background.rs [hunk 1]
```diff
@@ -353,6 +353,24 @@ impl BackgroundRunnerCore {
...
+                .range_manager()
+                .historical_ranges()
+                .iter()
+                .find_map(|(r, m)| {
+                    if r.contains_range(range) {
+                        Some(
+                            m.range_snapshot_list()
+                                .min_snapshot_ts()
+                                .unwrap_or(u64::MAX),
+                        )
```
Comment (glorv) [inline components/range_cache_memory_engine/src/background.rs:368]:
```suggestion
                        m.range_snapshot_list()
                            .min_snapshot_ts()
```

**Review by glorv — APPROVED** [review]
LGTM

**Review by overvenus — APPROVED** [review]

File: Cargo.lock [hunk 2]
```diff
@@ -3562,18 +3562,18 @@ checksum = "77af24da69f9d9341038eba93a073b1fdaaa1b788221b00a69bce9e762cb32de"
 
 [[package]]
 name = "openssl-src"
-version = "111.25.0+1.1.1t"
```
Comment (overvenus) [inline Cargo.lock:3565]: Any reason for changing openssl?

**Review by overvenus — CHANGES_REQUESTED** [review]
(Prevent merge)

**Review by LykxSassinator — APPROVED** [review]
LGTM

File: Cargo.lock (hunk 2 above)
Comment (SpadeA-Tang) [reply Cargo.lock]: reverted

Comment (overvenus) [reply Cargo.lock]: It has been fixed #17292

**Review by overvenus — APPROVED** [review]

**Pull request merged**